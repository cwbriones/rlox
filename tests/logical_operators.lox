// Truthiness
print 1 or 2; // expect: 1
print 1 and 2; // expect: 2
print "foo" or "bar"; // expect: foo
print "hello" and "world"; // expect: world
print nil and "falsey"; // expect: nil
print false or "falsey"; // expect: falsey

// Short-Circuit
var a = "before";
nil and (a = "after");
print a; // expect: before

var b;
(b = "before") or (b = "after");
print b; // expect: before
