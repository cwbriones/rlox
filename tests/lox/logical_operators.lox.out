1
2
foo
world
nil
falsey
before
before
